File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int total = 0;
        int MaxTotal = nums[0];

        for(int i : nums) {
            if(total < 0) {
                total = 0;
            }

            total += i;
            MaxTotal = Math.max(MaxTotal, total);
        }

        return MaxTotal;
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        total = 0
        maxT = nums[0]

        for i in range(len(nums)):
            total += nums[i]
            maxT = max(maxT, total)

            if total < 0:
                total = 0
        return maxT